### PR TITLE
task/06-routing

### PR DESCRIPTION
**Pull request by Allocene**
Acceptance Criteria for 1

- [x]  Mentor is added to pr
- [x]  AC copied to pr description
- [x]  Link to deployed version is added to pr
- [x]  LayoutComponent with a static header and footer, and an Outlet for dynamic content, is added.
- [x]  Two routes, AboutPage and ProductsPage, are added.
- [x]  The AboutPage is set as the index route.
- [x]  A route for invalid pages is added, displaying Page not found with a link to the AboutPage.

Acceptance Criteria for 1*

- [x]  The layout for ProductPage is added according to the design.
- [x]  An additional route for ProductPage with dynamic parameters :id is added.
- [x]  An additional request getProduct is implemented, using the product's id.

https://allocene.github.io/fe-react-2024/

### CODE REVIEW

File: src/components/PageNotFound/pagenf.module.css
Comment (Fenix4088) [inline src/components/PageNotFound/pagenf.module.css:1]: please rename this file
page-not-found.css

File: src/components/PageNotFound/pagenf.module.css [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+.nfp {
+  font-size: 18px;
+  margin-bottom: 50px;
+}
+
+.nfbutton {
+  font-size: 32px;
+  padding: 10px;
+  border-radius: 20px;
+  border: solid 1px #EF4934;
```
Comment (Fenix4088) [inline src/components/PageNotFound/pagenf.module.css]: use css variables for colors in your css files

File: src/components/CartContent/ThemeContext.tsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+interface ThemeContextProps {
+    isDarkMode: boolean;
+    onThemeChange: (isDarkMode: boolean) => void;
+}
+
+interface ThemeProviderProps {
+    children: ReactNode;
+}
+
+const ThemeContext = createContext<ThemeContextProps | undefined>(undefined);
```
Comment (Fenix4088) [inline src/components/CartContent/ThemeContext.tsx]: good practies initialise contexts with `null` or default value, try to use as little as possible of `undefined`'s

File: src/components/CartContent/ThemeContext.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+};
+
+export const ThemeProvider: React.FC<ThemeProviderProps> = ({ children }) => {
+    const [isDarkMode, setIsDarkMode] = useState<boolean>(false);
+
+    const handleThemeChange = () => {
+        setIsDarkMode(isDarkMode);
+    };
+
+    return <ThemeContext.Provider value={{ isDarkMode, onThemeChange: handleThemeChange }}>{children}</ThemeContext.Provider>;
```
Comment (Fenix4088) [inline src/components/CartContent/ThemeContext.tsx:35]:
```suggestion
    const handleThemeChange = () => setIsDarkMode(isDarkMode);
    return <ThemeContext.Provider value={{ isDarkMode, handleThemeChange }}>{children}</ThemeContext.Provider>;
```
or
```suggestion
    const onThemeChange = () => setIsDarkMode(isDarkMode);
    return <ThemeContext.Provider value={{ isDarkMode, onThemeChange }}>{children}</ThemeContext.Provider>;
```

this rename is usless

File: src/App.tsx [hunk 4]
```diff
@@ -1,36 +1,35 @@
...
-                <Header onPageChange={handlePageChange} onThemeChange={handleThemeChange} isDarkMode={isDarkTheme} />
-                {pageComponents[currentPage]}
-                <Footer />
-            </CartProvider>
-        </>
+        <Router basename="/fe-react-2024">
+            <ThemeProvider>
+                <CartProvider>
+                    <Routes>
+                        <Route path="/" element={<LayoutComponent onThemeChange={handleThemeChange} isDarkMode={isDarkTheme} />}>
```
Comment (Fenix4088) [inline src/App.tsx:23]: u created context for theme, so why u sending `isDarkTheme` as a property to the component, u case invoke context inside, and remove the comment

File: src/App.tsx [hunk 5]
```diff
@@ -1,36 +1,35 @@
...
+                            <Route index element={<About />} />
+                            <Route path="products" element={<Products isDarkTheme={isDarkTheme} />} />
```
Comment (Fenix4088) [inline src/App.tsx:25]: same question

**Review by Fenix4088 — APPROVED** [review]

File: src/App.tsx (hunk 4 above)
Comment (Allocene) [reply src/App.tsx:23]: I've been trying to fix this but I keep having something break and stop working, I can't figure out why... Help with this, plz

Comment (Fenix4088) [reply src/App.tsx:23]: okey, will check it out, in evening

Comment (Allocene) [reply src/App.tsx:23]: Thanks🙏🏼

**Pull request merged**